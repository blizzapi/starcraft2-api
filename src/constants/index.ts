export { profileUrlLocales as PROFILE_URL_LOCALES } from "./profileUrlLocales";
