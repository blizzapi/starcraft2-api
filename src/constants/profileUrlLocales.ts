/**
 * List of locales used on StarCraft II website (StarCraft2.com)
 */
const profileUrlLocales: ReadonlyArray<string> = [
  'en-us',
  'es-mx',
  'pt-br',
  'de-de',
  'en-gb',
  'es-es',
  'fr-fr',
  'it-it',
  'pl-pl',
  'ru-ru',
];

export default Object.freeze(profileUrlLocales);
