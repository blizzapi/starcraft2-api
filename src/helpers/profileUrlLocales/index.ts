export { checkIfProfileUrlLocaleLooksValid } from './checkIfProfileUrlLocaleLooksValid';
export { getAllProfileUrlLocales } from './getAllProfileUrlLocales';
export { profileUrlLocaleRegex } from './profileUrlLocaleRegex';
export { validateProfileUrlLocale } from './validateProfileUrlLocale';
