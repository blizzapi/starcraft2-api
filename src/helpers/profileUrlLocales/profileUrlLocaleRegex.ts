/**
 * Exports a regular expression matching profile URL locales (e.g. 'en-us')
 */
export default /^(?:[a-zA-Z]{2}-[a-zA-Z]{2})$/i;
