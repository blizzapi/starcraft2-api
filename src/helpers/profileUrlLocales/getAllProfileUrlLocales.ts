import { PROFILE_URL_LOCALES } from '../../constants';

/**
 * Returns a list of all profile URL locales (e.g. 'en-us')
 *
 * @return List of all available profile URL locales as array of strings.
 */
export const getAllProfileUrlLocales = () => PROFILE_URL_LOCALES;
