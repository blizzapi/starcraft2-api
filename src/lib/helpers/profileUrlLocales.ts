import constants from '../constants';

type ProfileUrlLocale = string;

/**
 * Returns a list of all profile URL locales
 *
 * @return List of all available profile URL locales as array of strings.
 */
export const getAllProfileUrlLocales = () => constants.PROFILE_URL_LOCALES;

/**
 * Verifies whether profile URL locale matches the regex pattern
 *
 * @param locale Profile URL locale
 * @return true if profile URL locale matches the pattern, false if not
 */
export const checkIfProfileUrlLocaleLooksValid = (profileUrlLocale: ProfileUrlLocale) => {
  const profileUrlLocaleRegex = /^(?:[a-z]{2}-[a-z]{2})$/gi;
  const doesProfileUrlLocaleLookValid = profileUrlLocaleRegex.test(profileUrlLocale);

  // tslint:disable-next-line: no-object-mutation
  profileUrlLocaleRegex.lastIndex = 0;
  return doesProfileUrlLocaleLookValid;
};

/**
 * Validates locale profile URL locale against profile URL locale list
 * (whether it exists in the constants object)
 *
 * @param locale Locale name
 * @return true if profile URL locale exists, false if not.
 * Throws RangeError if profile URL locale doesn't match regex pattern
 */
export const validateProfileUrlLocale = (profileUrlLocale: ProfileUrlLocale) => {
  const isProfileUrlLocaleValid = checkIfProfileUrlLocaleLooksValid(profileUrlLocale);
  const availableProfileUrlLocales = getAllProfileUrlLocales();
  const lowerCaseLocale = profileUrlLocale.toLowerCase();

  if (!isProfileUrlLocaleValid) {
    throw new RangeError(`${profileUrlLocale} is not a valid parameter for validateProfileUrlLocale()`);
  }

  return availableProfileUrlLocales.includes(lowerCaseLocale);
};
