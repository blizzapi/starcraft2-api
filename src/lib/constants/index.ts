import profileUrlLocales from './profileUrlLocales';

export default {
  PROFILE_URL_LOCALES: profileUrlLocales,
}