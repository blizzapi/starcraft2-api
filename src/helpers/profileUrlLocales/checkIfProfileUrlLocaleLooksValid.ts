import profileUrlLocaleRegex from './profileUrlLocaleRegex';

/**
 * Verifies whether profile URL locale matches regex pattern
 *
 * @param profileUrlLocale Profile URL locale (e.g. 'en-us')
 * @return true if profile URL locale matches the pattern, false if not
 */
export default (profileUrlLocale: string) =>
  profileUrlLocaleRegex.test(profileUrlLocale);
