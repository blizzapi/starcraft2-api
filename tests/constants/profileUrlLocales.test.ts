import { profileUrlLocales } from '../../src/constants/profileUrlLocales';

describe('profileUrlLocales', () => {
  it('should match snapshot', () => {
    expect(profileUrlLocales).toMatchSnapshot();
  });
});
