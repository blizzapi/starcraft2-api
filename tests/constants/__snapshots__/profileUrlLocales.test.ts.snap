// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profileUrlLocales should match snapshot 1`] = `
[
  "en-us",
  "es-mx",
  "pt-br",
  "de-de",
  "en-gb",
  "es-es",
  "fr-fr",
  "it-it",
  "pl-pl",
  "ru-ru",
]
`;
